src/backend/distributed/commands/call.c

```diff
@@ -142,12 +148,26 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 		return false;
 	}
 
+
+#if PG_VERSION_NUM >= PG_VERSION_14
+
+	/*
+	 * We might need to add outargs to the funcExpr->args so that they can
+	 * be pushed down. We can implement in the future.
```

Comment:
Is this a limitation in the deparsing logic?

Reply:
I think the main issue is that Postgres represents out parameters in `Call` statement in an unusual way.  

```C
/* ----------------------
 *		CALL statement
 *
 * OUT-mode arguments are removed from the transformed funcexpr.  The outargs
 * list contains copies of the expressions for all output arguments, in the
 * order of the procedure's declared arguments.  (outargs is never evaluated,
 * but is useful to the caller as a reference for what to assign to.)
 * ----------------------
 */
typedef struct CallStmt
{
	NodeTag		type;
	FuncCall   *funccall;		/* from the parser */
	FuncExpr   *funcexpr;		/* transformed call, with only input args */
	List	   *outargs;		/* transformed output-argument expressions */
} CallStmt;

```

So, the input (and inout) parameters are listed in `funccall->args` and the out parameters are listed in `call->outargs`.

We probably need to find a way to merge these two lists so that we update `func->args` including the `outargs` and deparser adds them to the string.

The challenge was that `funccall->args` contains both `IN` and `INOUT` parameters, and `outargs` `INOUT` and `OUT` arguments. Which means that it is not trivial to merge to lists, so for now I skipped after spending 2+ hours on it :)

Reply:
Got it, I guess we'll need dedicated deparsing logic for CALL statements.